protected/humhub/modules/file/models/File.php

```diff
@@ -240,24 +313,60 @@ public function canDelete($userId = null)
      */
     public function isAssigned()
     {
-        return ($this->object_model != "");
+        return (!empty($this->object_model));
     }
 
     /**
      * Checks if this file is attached to the given record
+     *
      * @param ActiveRecord $record
+     *
      * @return bool
      */
     public function isAssignedTo(ActiveRecord $record)
     {
         return $this->object_model === get_class($record) && $this->object_id == $record->getPrimaryKey();
     }
 
+    /**
+     * @return Metadata
+     */
+    public function getMetadata(): Metadata
+    {
+        /** @var Metadata|null $md */
+        $md = $this->getAttribute('metadata');
+
+        if ($md instanceof Metadata) {
+            return $md;
+        }
+
+        $md = new Metadata($md);
+
+        $this->setAttribute('metadata', $md);
+
+        return $this->metadata;
+    }
+
+    /**
+     * @param string|array $metadata
+     *
+     * @return File
+     */
+    public function setMetadata($metadata): File
```

Comment:
TBD

Reply:
Alternatively, this could be removed entirely and make the property read-only.

- To set a value, one would need to do:
   ``` php
   $model->getMetadata()->property = 'value';
   $model->metadata->property = 'value';
   ```
   You find more examples here:
   https://github.com/humhub/humhub/blob/78df7d89e17ac95106fd7afa41ed326f6dd61df8/protected/humhub/modules/file/models/File.php#L51-L62
- With the current implementation of `setMetadata()`, the above could be like so:
   ``` php
   $model->setMetadata('property' => 'value');
   
   // this would delete all the data within the metadata object
   $model->setMetadata(null);
   $model->metadata = null;
   ```